euth/comments/serializers.py

```diff
@@ -31,11 +34,42 @@ def get_child_comments(self, obj):
         pk = obj.pk
         children = Comment.objects.all().filter(
             content_type=content_type, object_pk=pk).order_by('created')
-        serializer = CommentSerializer(children, many=True)
+        serializer = CommentSerializer(
+            children,
+            many=True,
+            context={'request': self.context['request']})
         return serializer.data
 
     def get_is_deleted(self, obj):
         """
         Returns true is one of the flags is set
         """
         return (obj.is_censored or obj.is_removed)
+
+    def get_rates(self, obj):
+        """
+        Gets positve and negative rate count as well as
+        info on the request users rate
+        """
+        user = self.context['request'].user
+        contenttype = ContentType.objects.get_for_model(obj)
+        obj_rates = rate_models.Rate.objects.filter(
+            content_type=contenttype, object_pk=obj.pk)
+        positive_rates = obj_rates.filter(value=1).count()
+        negative_rates = obj_rates.filter(value=-1).count()
+        try:
+            user_rate = obj_rates.get(user=user)
+            user_rate_value = user_rate.value
+            user_rate_id = user_rate.pk
+        except:
+            user_rate_value = None
+            user_rate_id = None
+
+        result = {
+            'positive_rates': positive_rates,
+            'negative_rates': negative_rates,
+            'current_user_rate_value': user_rate_value,
+            'current_user_rate_id': user_rate_id
+        }
+
+        return result
```

Comment:
I would suggest to change the code according to my proposal in `euth/rates/serializers.py`.